packages/twenty-server/src/engine/metadata-modules/object-metadata/services/object-metadata-migration.service.ts

```diff
@@ -264,4 +267,44 @@ export class ObjectMetadataMigrationService {
       ],
     );
   }
+
+  public async recomputeEnumNames(
+    existingObjectMetadata: ObjectMetadataEntity,
+    updatedObjectMetadata: ObjectMetadataEntity,
+    workspaceId: string,
+  ) {
+    const fieldMetadataToUpdate = await this.fieldMetadataRepository.find({
+      where: {
+        objectMetadataId: updatedObjectMetadata.id,
+        workspaceId,
+        type: In([
+          FieldMetadataType.SELECT,
+          FieldMetadataType.MULTI_SELECT,
+          FieldMetadataType.RATING,
+          FieldMetadataType.ACTOR,
+        ]),
+      },
+    });
+
+    for (const fieldMetadata of fieldMetadataToUpdate) {
+      await this.workspaceMigrationService.createCustomMigration(
+        generateMigrationName(`update-${fieldMetadata.name}-enum-names`),
+        workspaceId,
+        [
+          {
+            name: computeTableName(
+              updatedObjectMetadata.nameSingular,
+              updatedObjectMetadata.isCustom,
+            ),
+            action: WorkspaceMigrationTableActionType.ALTER,
+            columns: this.workspaceMigrationFactory.createColumnActions(
+              WorkspaceMigrationColumnActionType.ALTER,
+              fieldMetadata,
```

Comment:
kinda hacky there where we pass the same fieldMetadata for existing and new since this is not the field that has changed but we still want to trigger the whole flow